frontend/www/js/omegaup/components/course/List.test.ts

```diff
@@ -0,0 +1,79 @@
+import { shallowMount } from '@vue/test-utils';
+import expect from 'expect';
+import Vue from 'vue';
+
+import T from '../../lang';
+import { omegaup } from '../../omegaup';
+import { types } from '../../api_types';
+
+import course_List from './List.vue';
+
+const coursesListProps = {
+  courses: {
```

Comment:
```suggestion
  courses: <types.AlgunoTipoBonito>{
```

eso implica que también hay que cambiar https://github.com/omegaup/omegaup/blob/1dd8c5b61b397b9d8c54a097cbe13669dfe05e2e/frontend/www/js/omegaup/components/course/List.vue#L44 para no tener el tipo incorrecto (!!!) y que https://github.com/omegaup/omegaup/blob/1dd8c5b61b397b9d8c54a097cbe13669dfe05e2e/frontend/server/src/Controllers/Course.php#L26 sea `array{courses: AlgunTipoBonito}`, `AlgunTipoBonito` es `array{admin: Algo, public: Algo, student: Algo}` y _ese_ tipo `Algo` es `array{accessMode: string, activeTab: string, filteredCourses: array{current: array{courses: list<FilteredCourse>, timeType: string}, past: array{courses: list<FilteredCourse>, timeType: string}}}`.

Reply:
Listo, reutilicé las anotaciones de #4344 para no tener tipos duplicados luego

Reply:
ándale, esa es una mejor idea!